sys/net/gnrc/link_layer/netdev2/gnrc_netdev2_ieee802154.c

```diff
@@ -122,17 +127,14 @@ static gnrc_pktsnip_t *_recv(gnrc_netdev2_t *gnrc_netdev2)
                   gnrc_netif_addr_to_str(src_str, sizeof(src_str),
                                          gnrc_netif_hdr_get_src_addr(hdr),
                                          hdr->src_l2addr_len),
-                  nread);
+                  nread - mhr_len);
 #if defined(MODULE_OD)
-            od_hex_dump(pkt->data, nread, OD_WIDTH_DEFAULT);
+            od_hex_dump(pkt->data, nread - mhr_len, OD_WIDTH_DEFAULT);
 #endif
 #endif
             gnrc_pktbuf_remove_snip(pkt, ieee802154_hdr);
             LL_APPEND(pkt, netif_hdr);
         }
-
-        DEBUG("_recv_ieee802154: reallocating.\n");
-        gnrc_pktbuf_realloc_data(pkt, nread);
```

Comment:
I'm not sure, but I have the glimpse of a memory that there was a reason why I moved it here (as compared to `gnrc_netdev2_eth`) ⇒ this needs intensive testing before merging.